### ISSUE TITLE
How to close session on hardware?

### ISSUE THREAD

**Issue by ferhat00** [AlexShkarin/pyLabLib#1]
Thank you for the great work you are doing on this. 

I have a question regarding using the Thorlabs FW class. Is there a way to close a connection? At the moment I have to close my python kernel and open it again every time I want to run my script top to bottom.

I get the error message:
BackendOpenError: could not open port 'COM5': PermissionError(13, 'Access is denied.', None, 5)

When I close my IDE/python kernel, and then re-open it, it connects and works fine.

Many thanks in advance.

**Comment by AlexShkarin** [comment]
Indeed, in Windows access to COM-ports is non-shareable, and connection is closed either when the running process is terminated (e.g., when you close your Python kernel), or when the resource is explicitly closed. 

For the latter, all device classes have `.close()` method, which closes device connection. No device commands can be executed afterwards, but the connection can be reopened by calling `.open()` method or by re-creating the device object. Hence, you just need to put `dev.close()` in the end of your script to close the connection.

Alternatively, you can use Python `with` statement to handle this automatically, e.g.:
```python
with Thorlabs.FW("COM5") as dev:
    dev.set_position(6)
```
In this example the connection is automatically closed on the exit from the `with` block.

**Comment by ferhat00** [comment]
Many thanks for your suggestions. I will implement it.
